### ISSUE TITLE
Implement Shadowfile

### ISSUE THREAD

**Issue by MggMuggins** [redox-os/users#14]
This is something I've been thinking of for a while. Right now the crate allows users to authenticate themselves, which is a serious no-no. My main thoughts on implementation would be to completely remove the hash field in `/etc/passwd` and add `/etc/shadow` with semi-colon delimited key-value pairs (username to hash), like this:
```
user;!
```
In API-land, this would probably only require adding a parameter to `AllUsers::new()`, a boolean to see if the called wants to be able to authenticate or not. AllUsers can then initialize and read files accordingly.

**Comment by MggMuggins** [comment]
@jackpot51 @goyox86 Like I said, I want the password field completely not in `/etc/passwd`. Is it a priority at all to preserve existing installs, and if so, is there a good way to implement this without a password field in `/etc/passwd`?

**Comment by jackpot51** [comment]
No, it is not a priority to preserve the format of /etc/passwd. Just make sure the correct edits are done in https://github.com/redox-os/installer

**Comment by MggMuggins** [comment]
Yeah, I'll make sure that happens before I merge anything into master. Although since this is now up on crates I'll make changes here and then we'll do a big release and change the installer.
